ekncontent/ekncontent/eknc-domain.c

```diff
@@ -830,8 +858,13 @@ on_xapian_query_response (GObject *source,
           return;
         }
 
+      /* Note that we pass another closure here instead of just the task
```

Comment:
Hmm, I see that now, but I'm still picking up a code smell from this. Maybe create a new task for these operations then, with its own task data that's not shared with `QueryState`?